modules/vba_documents/spec/lib/upload_validator_spec.rb

```diff
@@ -62,6 +62,44 @@
         end
       end
     end
+
+    context 'when vba_documents.custom_metadata_allow_list Setting is not accessable(nil)' do
+      let(:included_values) { { unpermitted_key: 'ahhh!!!!' } }
+
+      it 'logs missing\invalid setting warning' do
+        expect(Rails.logger).to receive(:warn).with(
+          'VBADocuments missing or malformed Setting: custom_metadata_allow_list'
+        )
+        expect { output }.not_to raise_error
+      end
+    end
+
+    context 'when unpermitted metadata is supplied' do
+      let(:included_values) { { unpermitted_key: 'ahhh!!!!' } }
+
+      it 'logs unpermitted metadata warning' do
+        with_settings(Settings.vba_documents, custom_metadata_allow_list: { other_consumer_id: ['key3'] }.to_json) do
+          expect(Rails.logger).to receive(:warn).with(
+            'VBADocuments unpermitted metadata supplied. Guid: guid Keys: unpermitted_key'
+          )
+          expect { output }.not_to raise_error
+        end
+      end
+    end
+
+    context 'when consumer specific metadata is supplied' do
+      let(:included_values) { { unpermitted_key: 'ahhh!!!!' } }
+
+      it 'does not log' do
+        with_settings(Settings.vba_documents,
+                      custom_metadata_allow_list: { consumer_id: ['unpermitted_key'] }.to_json) do
+          expect(Rails.logger).not_to receive(:warn).with(
+            'VBADocuments unpermitted metadata supplied. Guid: guid Keys: unpermitted_key'
+          )
```

Comment:
in this case we check to make sure the warning is NOT logged since we stubbed in the extra metadata into the allow setting